### PR TITLE
Replace Travis CI with GitHub Actions

### PR DESCRIPTION
**Pull request by hugovk**
Travis CI has a new pricing model which places limits on open source.

* https://blog.travis-ci.com/2020-11-02-travis-ci-new-billing
* https://www.jeffgeerling.com/blog/2020/travis-cis-new-pricing-plan-threw-wrench-my-open-source-works

Many projects are moving to GitHub Actions instead, including Jazzband projects: https://github.com/jazzband-roadies/help/issues/206

Tests are already run on GHA, this PR removes the Travis CI config and adds a deploy workflow. This is based on the one we already got working in https://github.com/jazzband/prettytable. The release process will be the same.

TODO:

* [x] @jezdez to add `JAZZBAND_RELEASE_KEY` to the repo secrets.
* [x] @jezdez Remove "continuous-integration/travis-ci Expected — Waiting for status to be reported" from branch protection rules.

**Changelog-friendly one-liner**: Replace Travis CI with GitHub Actions

##### Contributor checklist

- [x] Provided the tests for the changes.
- [x] Gave a clear one-line description in the PR (that the maintainers can add to CHANGELOG.md on release).
- [x] Assign the PR to an existing or new milestone for the target version (following [Semantic Versioning](https://blog.versioneye.com/2014/01/16/semantic-versioning/)).

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+          twine check dist/*
+
+      - name: Publish package to Jazzband
+        if: github.event.action == 'published'
+        uses: pypa/gh-action-pypi-publish@master
+
+        with:
+          user: jazzband
+          password: ${{ secrets.JAZZBAND_RELEASE_KEY }}
+          repository_url: https://jazzband.co/projects/prettytable/upload
```
Comment (jezdez) [inline .github/workflows/deploy.yml:52]: Needs to be fixed.

**Comment by atugushev** [comment]
Thank you @hugovk! You are right on time! 🙏 


```yaml
uses: deadsnakes/action@v2.0.1
```

FYI it fixex 3.9-dev job.

**Comment by atugushev** [comment]
@jezdez could you please also remove the "required" status check from Travis CI?

**Comment by jezdez** [comment]
I've added the `JAZZBAND_RELEASE_KEY` secret.

**Comment by jezdez** [comment]
> @jezdez could you please also remove the "required" status check from Travis CI?

Done. I've left appveyor though.

@davidovich @vphilippon @atugushev There is a large number of test envs on GitHub actions, should any be required to pass first before a merge is happening?

**Comment by jezdez** [comment]
BTW, GitHub Actions supports Windows as well, do we want to get rid of Appveyor as well?

**Comment by atugushev** [comment]
> Done. I've left appveyor though.

@jezdez Thanks! I think appveyor also is a candidate to remove from CI (and so required status check).


> There is a large number of test envs on GitHub actions, should any be required to pass first before a merge is happening?
I would say:

- CI / Ubuntu / 3.8 / latest
- CI / Windows / 3.8 / latest
- CI / macOS / 3.8 / latest

**Comment by jezdez** [comment]
> > Done. I've left appveyor though.
> 
> @jezdez Thanks! I think appveyor also is a candidate to remove from CI (and so required status check).

Okay, I'd support that. I see there are already Windows runs set-up. Can I just disable AppVeyor? Let's at least open an issue to track this if non of the other project leads agree to do that. There is a [pretty old issue](https://github.com/jazzband-roadies/help/issues/74) about the lack of access for Jazzband members as well.

> > There is a large number of test envs on GitHub actions, should any be required to pass first before a merge is happening?
> 
> I would say:
> 
>     * CI / Ubuntu / 3.8 / latest
> 
>     * CI / Windows / 3.8 / latest
> 
>     * CI / macOS / 3.8 / latest

Done.

**Comment by atugushev** [comment]
> Can I just disable AppVeyor? Let's at least open an issue to track this if non of the other project leads agree to do that. There is a [pretty old issue](https://github.com/jazzband-roadies/help/issues/74) about the lack of access for Jazzband members as well.

Yeah, let's disable it first.

**Review by atugushev — APPROVED** [review]
LGTM 👍

**Comment by atugushev** [comment]
Thanks @hugovk and @jezdez!

**Pull request merged**